src/Features/Core/Portable/InlineHints/AbstractInlineParameterNameHintsService.cs

```diff
@@ -32,21 +35,130 @@ public async Task<ImmutableArray<InlineParameterHint>> GetInlineParameterNameHin
             if (!literalParameters && !objectCreationParameters && !otherParameters)
                 return ImmutableArray<InlineParameterHint>.Empty;
 
+            var hideForParametersThatDifferOnlyBySuffix = options.GetOption(InlineHintsOptions.HideForParametersThatDifferOnlyBySuffix);
+            var hideForParametersThatMatchMethodIntent = options.GetOption(InlineHintsOptions.HideForParametersThatMatchMethodIntent);
```

Comment:
I like "match method intent". Perhaps "when parameter name matches the method intent" might be better text for the option in the UI too. "clear based on method name" is quite broad.